### PR TITLE
feat(shwap/store): Add q4 trimming support in store

### PR DESCRIPTION
**Pull request by walldiss**

### CODE REVIEW

File: store/metrics.go [hunk 1]
```diff
@@ -130,15 +138,27 @@ func (m *metrics) observeHas(ctx context.Context, dur time.Duration, failed bool
 		attribute.Bool(failedKey, failed)))
 }
 
-func (m *metrics) observeRemove(ctx context.Context, dur time.Duration, failed bool) {
+func (m *metrics) observeRemoveAll(ctx context.Context, dur time.Duration, failed bool) {
+	if m == nil {
+		return
+	}
+	if ctx.Err() != nil {
+		ctx = context.Background()
```
Comment (cristaloleg) [inline store/metrics.go:146]: So, always blocking? Maybe `context.WithTimeout(context.Background(), time.Second)` ?

Comment (walldiss) [reply store/metrics.go:146]: Good that you pointed out. We use common function to reset context in metrics everywhere else. I'll change how we do it in another PR.

File: store/store.go [hunk 2]
```diff
@@ -305,51 +304,76 @@ func (s *Store) hasByHeight(height uint64) (bool, error) {
 	return exists(pathODS)
 }
 
-func (s *Store) Remove(ctx context.Context, height uint64, datahash share.DataHash) error {
+func (s *Store) RemoveAll(ctx context.Context, height uint64, datahash share.DataHash) error {
```
Comment (Wondertan) [inline store/store.go:307]: Let's name to `RemoveODSQ4`. Its not clear what All stands for. 

Same for helpers methods and metrics.

File: store/store.go [hunk 3]
```diff
@@ -33,7 +33,7 @@ const (
 var ErrNotFound = errors.New("eds not found in store")
 
 // Store is a storage for EDS files. It persists EDS files on disk in form of Q1Q4 files or ODS
-// files. It provides methods to put, get and remove EDS files. It has two caches: recent eds cache
+// files. It provides methods to put, get and removeAll EDS files. It has two caches: recent eds cache
```
Comment (Wondertan) [inline store/store.go:36]: rename relic

**Pull request merged**